src/featurelistmodelhighlight.cpp

```diff
@@ -83,14 +91,18 @@ QSGNode* FeatureListModelHighlight::updatePaintNode( QSGNode* n, QQuickItem::Upd
       QModelIndex index = mModel->index( i, 0, QModelIndex() );
       const Feature& feature = mModel->data( index, FeatureListModel::FeatureRole ).value<Feature>();
 
+      QgsFeature feat( feature.qgsFeature() );
```

Comment:
No reason for a complete feature copy.

    QgsGeometry geom( feature.constGeometry() );
    geom.transform( transf );
    ....
    sn = new QgsSGGeometry( geom, mSelectionColor, mWidth );

should be ok.